oauth2_http/java/com/google/auth/oauth2/IdentityPoolCredentials.java

```diff
@@ -136,7 +139,7 @@ public static class Builder extends ExternalAccountCredentials.Builder {
 
     Builder(IdentityPoolCredentials credentials) {
       super(credentials);
-      if (credentials.subjectTokenProvider.isUserSupplied()) {
+      if (credentials.builtWithSupplier) {
```

Comment:
If the above change is made, we don't need this here (also, probably don't need to expose supplier at all).